lib/stack.rb

```diff
@@ -1,21 +1,24 @@
+require_relative './linked_list.rb'
+
 class Stack
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = LinkedList.new
   end
-
+  
   def push(element)
-    raise NotImplementedError, "Not yet implemented"
+    @store.add_last(element)
+    return @store
   end
-
+  
   def pop
-    raise NotImplementedError, "Not yet implemented"
+    return nil if @store.get_first == nil
+    return @store.remove_last
   end
-
+  
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return true if @store.get_first == nil
```

Comment:
Just a note that LinkedList has a `.empty?` method.

```suggestion
    return @store.empty?
```